### PR TITLE
Add Australia state switcher

### LINKED ISSUE #6177: Australia state switcher on DCR fronts

### LINKED ISSUE #6234: Support Aus region selector on DCR
Containers with the 'targetedTerritory' set to an Australian region (Queensland, Victoria, New south wales) should receive the region selector component below them: 

image

Frontend implementation PR: https://github.com/guardian/frontend/pull/25361

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Add the australian state switcher on DCR fronts, which is required to display localised news on the `AU` front.

Once this is done, we can [update Frontend’s `FaciaPicker` to allow displaying fronts with Australian territories targetted containers](https://github.com/guardian/frontend/blob/0ab174b0b3982a68c60338130c6b68ed58b2e75c/facia/app/services/dotcomrendering/FaciaPicker.scala#L127-L138).

Pass local cookies to the content request, which enables local testing of the targetted territories feature.

## Why?

Closes #6177 
Closes #6234

## Screenshots

We can see slight differences between the two rendering tiers, but I deemed them acceptable as the DCR version is using native **Source** components rathen than an exact visual replica.

| Frontend      | DCR      |
| ------------- | ---------- |
| ![Frontend][] | ![DCR][] |
| ![Frontend mobile][] | ![DCR mobile][] |

[Frontend]: https://github.com/guardian/dotcom-rendering/assets/76776/7d371249-0584-4f3f-96d3-06dcd4d52d07
[DCR]: https://github.com/guardian/dotcom-rendering/assets/76776/b5991372-f4d5-4453-9d25-9235f767c596

[Frontend mobile]: https://github.com/guardian/dotcom-rendering/assets/76776/e29fef95-d1c0-45d8-bb25-62870777c776
[DCR mobile]: https://github.com/guardian/dotcom-rendering/assets/76776/e516f4e8-25d3-4fad-aaca-d9a0980a5d2d

**Note:** Compared to Frontend, we removed “Hide local news”, as hiding the container is sufficient for this use-case, and when people click this button they cannot get the local news container ever again–bar modifying their cookies manually. Discussed with @rhiannareechaye

### CODE REVIEW

**Review by AshCorr — APPROVED** [review]
🇦🇺 🥳 

Nice! Do we need to get Alex to sign off on the design changes? They do look significantly different.

File: dotcom-rendering/src/types/front.ts [hunk 1]
```diff
@@ -333,6 +335,7 @@ export type FECollectionType = {
 	showLatestUpdate: boolean;
 	config: FECollectionConfigType;
 	hasMore: boolean;
+	targetedTerritory?: AustralianTerritory;
```
[the remark this answers is not in the document]
Comment (mxdvl) [reply dotcom-rendering/src/types/front.ts]: Great catch, addressed now the addition of a [`territory.ts`](https://github.com/guardian/dotcom-rendering/pull/7984/files#diff-d9e26e4c6d6ed34387f2089ae2c8fff033d945a39f97d55dd92cfc9f9804ec1b)

File: dotcom-rendering/src/components/AustralianTerritorySwitcher.importable.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	'AU-QLD': 'Queensland',
+	'AU-VIC': 'Victoria',
+} as const satisfies Record<AustralianTerritory, string>;
+
+const controls = 'change-territory';
+
+type Props = {
+	targetedTerritory: AustralianTerritory;
+};
+export const AustralianTerritorySwitcher = ({ targetedTerritory }: Props) => {
```
Comment (OllysCoding) [inline dotcom-rendering/src/components/AustralianTerritorySwitcher.importable.tsx:30]: Curious what happens in a no-js experience - I don't think there's a lot we can do but perhaps worth having it written down in this PR.

File: dotcom-rendering/src/components/FrontSection.tsx [hunk 3]
```diff
@@ -549,7 +555,14 @@ export const FrontSection = ({
 			</div>
 
 			<div css={[sectionContentPadded, sectionShowMore, bottomPadding]}>
-				{showMore && (
+				{isString(targetedTerritory) &&
+				isAustralianTerritory(targetedTerritory) ? (
+					<Island>
```
Comment (OllysCoding) [inline dotcom-rendering/src/components/FrontSection.tsx:560]: Can this be delayed until visible or even until interaction?

**Review by OllysCoding — APPROVED** [review]
+1 Looks good to me :)

File: dotcom-rendering/src/components/AustralianTerritorySwitcher.importable.tsx (hunk 2 above)
Comment (mxdvl) [reply dotcom-rendering/src/components/AustralianTerritorySwitcher.importable.tsx:30]: I guess we could make this a client-only island…

Comment (mxdvl) [reply dotcom-rendering/src/components/AustralianTerritorySwitcher.importable.tsx:43]: Went for `onInteraction` in the end

**Comment by mxdvl** [comment]
Well, hello there!

image

**Pull request merged**